src/audio/soundengine.cpp

```
@@ -169,4 +180,84 @@ void SoundEngine::setListenerGain(float gain)
 {
   m_device->setListenerGain(gain);
 }
+
+void SoundEngine::registerStream(size_t slot,
+                                 const gsl::shared_ptr<StreamVoice>& stream,
+                                 const std::filesystem::path& path)
+{
+  m_streams.emplace(slot, StreamInfo{stream, path});
```

This will silently fail when the slot is already in use.